src/librustc/ty/sty.rs

```diff
@@ -2008,12 +2008,12 @@ impl<'tcx> Const<'tcx> {
 
     #[inline]
     pub fn from_bool(tcx: TyCtxt<'_, '_, 'tcx>, v: bool) -> &'tcx Self {
-        Self::from_bits(tcx, v as u128, ParamEnv::empty().and(tcx.types.bool))
+        Self::from_bits(tcx, v as u128, ParamEnv::reveal_all().and(tcx.types.bool))
```

Comment:
@eddyb won't like that, they taught me that `reveal_all` is usually wrong. I don't have any clue what this does, though, so I'll defer to your judgement here @oli-obk.

Reply:
I can change it back. But it's actually irrelevant here, since it gets used in `layout_of` which (I checked) directly turns any `ParamEnv` into a `reveal_all`.

Reply:
> directly turns any ParamEnv into a reveal_all.

Oh. Is there a way to make it not take a `ParamEnv` then if it gets ignored anway?